### PR TITLE
Fixes for top-of-tree Halide

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

File: src/JITModule.cpp [hunk 1]
```diff
@@ -334,7 +334,12 @@ void JITModule::compile_module(std::unique_ptr<llvm::Module> m, const string &fu
     // Build TargetMachine
     llvm::orc::JITTargetMachineBuilder tm_builder(llvm::Triple(m->getTargetTriple()));
     tm_builder.setOptions(options);
+#if LLVM_VERSION >= 180
+    tm_builder.setCodeGenOptLevel(CodeGenOptLevel::Aggressive);
+#else
     tm_builder.setCodeGenOptLevel(CodeGenOpt::Aggressive);
+#endif
+    tm_builder.setCodeGenOptLevel(opt_level);
```
Comment (abadams) [inline src/JITModule.cpp:342]: I'm confused by this change. It calls tm_builder.setCodeGenOptLevel twice?

File: src/LLVM_Output.cpp [hunk 2]
```diff
@@ -425,11 +425,20 @@ std::unique_ptr<llvm::Module> compile_module_to_llvm_module(const Module &module
 }
 
 void compile_llvm_module_to_object(llvm::Module &module, Internal::LLVMOStream &out) {
+#if LLVM_VERSION >= 180
+    emit_file(module, out, llvm::CodeGenFileType::ObjectFile);
+#else
     emit_file(module, out, llvm::CGFT_ObjectFile);
+#endif
+    emit_file(module, out, file_type);
```
Comment (abadams) [inline src/LLVM_Output.cpp:433]: Same confusion here

File: src/JITModule.cpp (hunk 1 above)
Comment (steven-johnson) [reply src/JITModule.cpp]: Bah, bonehead copy/paste error

File: src/LLVM_Output.cpp (hunk 2 above)
Comment (steven-johnson) [reply src/LLVM_Output.cpp]: And more bonehead

**Review by abadams — APPROVED** [review]

**Pull request merged**